csv-core/src/reader.rs

```diff
@@ -913,106 +926,105 @@ impl Reader {
         }
     }
 
+
     /// Compute the next NFA state given the current NFA state and the current
     /// input byte.
     ///
-    /// This returns the next NFA state along with two booleans that indicate
-    /// whether an input byte was consumed (i.e., a non-epsilon transition)
-    /// and whether the input byte should be copied to a caller provided output
-    /// buffer. For example, seeing a field delimiter will consume an input
-    /// byte but should not copy it to the output buffer.
+    /// This returns the next NFA state along with an NfaInputAction that
+    /// indicates what should be done with the input byte (nothing for an epsilon
+    /// transition, copied to a caller provided output buffer, or discarded).
     ///
-    /// It is illegal for this to emit an output byte without consuming the
-    /// input byte.
+    /// Note that this is more like a DFA than a true NFA; you cannot transition
+    /// to multiple states.
```

Comment:
I disagree with this statement. The presence of epsilon transitions puts this firmly in NFA territory.